userland/libtock/crt0.c

```diff
@@ -61,32 +63,33 @@ void _start(void* text_start,
   }
 
   // fix up GOT
+  uint32_t text_start = (uint32_t)app_start + myhdr->text_offset;
   volatile uint32_t* got_start     = (uint32_t*)(myhdr->got_start + stacktop);
-  volatile uint32_t* got_sym_start = (uint32_t*)(myhdr->got_sym_start + (uint32_t)text_start);
+  volatile uint32_t* got_sym_start = (uint32_t*)(myhdr->got_sym_start + (uint32_t)app_start);
   for (uint32_t i = 0; i < (myhdr->got_size / (uint32_t)sizeof(uint32_t)); i++) {
```

Comment:
I run into aliasing issues here on nrf51 / cortex m0.
When I am here in when starting the sensors app, stacktop is 0x20002470, myhdr->got_start is 0, and so got_start begins with the value 0x20002470. But &got_start is 0x20002474, so the second iteration of the loop overwrites got_start and all following got updates end up somewhere in the woods.

Removing the definition of text_start and inlining the code prevents got_start from being spilled from a register to RAM and makes the code work.

Reply:
Well, got_start is at sp + 4, which is where you expect it until line 61, after which sp + 4 is 0x20002474. 

Changing sp and r9 should probably be the last thing before calling main().